### DISCUSSION TITLE
Best practices with the wdio intercept service

### DISCUSSION THREAD

**Discussion by chadbellangivelify** [webdriverio-community/wdio-intercept-service#436] [Q&A]
I wanted to inquire:
1. If I have multiple tests in a test file that use the intercept service, should I call browser.setupInterceptor() within each test or do you suggest that I run the browser.setupInterceptor() once in the before

2. When do you suggest that I disable the interceptor:
a) within each test when the test is done
b) not at all and let the browser shutdown take care of it (does shutting down the browser actually take care of the network interceptor teardown
c) or once at the very end of all the tests (using the after)

**Comment by qatestingresearcher** [comment]
@chadbellangivelify : Did you get any updates on the same?

**Comment by erwinheitzman** [comment]
You run the method when you want to start capturing network requests.
You disable when you want to stop (refresh the state) or if you don't care, you can leave it till the instance is closed.
I'm not aware of any performance impact but will say that I believe that this is probably the last place you should look to gain a performance increase if any can be had at all.

EDIT: with the above, I mean to say that it does not matter when you start stop although generally I would only capture the traffic when needed and not always capture everything by default.

**Reply by qatestingresearcher** [reply]
Thanks @erwinheitzman

**Comment by chmanie** [comment]
I took the liberty to convert this to a Discussion :)
